aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -865,7 +958,22 @@ module AptosFramework::Stake {
         };
 
         let rewards_amount = distribute_reward(&mut stake_pool.active, num_blocks, num_successful_votes, remaining_lockup_time, validator_set_config);
-        rewards_amount = rewards_amount + distribute_reward(&mut stake_pool.pending_inactive, num_blocks, num_successful_votes, remaining_lockup_time, validator_set_config);
+        // Only give out rewards for pending_inactive if the total stake (active + pending_inactive) doesn't exceed the
+        // max allowed.
+        // First, this ensures rewards are only given to the mininum of (total stake, max stake allowed) as the voting
+        // power of a validator cannot exceed the max allowed (the excess stake above the max is effectively idle).
+        // Second, this prevents a potential manipulation where the validator withdraws the excess stake in order to
+        // receive rewards on both active and pending_inactive as the amounts individually do not exceed the max.
+        if (Coin::value(&stake_pool.active) + Coin::value(&stake_pool.pending_inactive) <= validator_set_config.maximum_stake) {
```

Comment:
hmm, this seems counter-intuitive, I thought they only receive rewards for the maximum stake regardless of how much is being withdrawn

Reply:
This does exactly that. It's just an implementation details - we distribute the rewards for active and pending_inactive separately (since those are 2 different Coin objects).

Reply:
I moved this calculation out to before the reward distribution and named it total_stake_before_rewards to be clearer.